packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/__tests__/quanticGeneratedAnswer.test.js

```diff
@@ -558,6 +560,88 @@ describe('c-quantic-generated-answer', () => {
+          describe('when the value is within the valid range', () => {
+            it('should set the answer height with the custom value', async () => {
+              mockAnswerHeight = defaultAnswerHeight - 25;
+              const expectedAnswerHeightValue = 300;
+              const element = createTestComponent({
+                ...defaultOptions,
+                collapsible: true,
+                maxCollapsedHeight: expectedAnswerHeightValue,
+              });
+              await flushPromises();
+
+              const generatedAnswer = element.shadowRoot.querySelector(
+                selectors.generatedAnswer
+              );
+              expect(generatedAnswer).not.toBeNull();
+
+              expect(
+                generatedAnswer.classList.contains(
+                  'generated-answer__answer--collapsed'
+                )
+              ).toBe(false);
+              // expect(generatedAnswer.style.maxHeight).toEqual(
+              //   `${expectedAnswerHeightValue}px`
+              // );
+            });
+          });
+
+          describe('when the value is greater than the valid range', () => {
+            it('should set the answer height with the fallback default value', async () => {
+              mockAnswerHeight = defaultAnswerHeight + 100;
+              const element = createTestComponent({
+                ...defaultOptions,
+                collapsible: true,
+                maxCollapsedHeight: 550,
+              });
+              await flushPromises();
+
+              const generatedAnswer = element.shadowRoot.querySelector(
+                selectors.generatedAnswer
+              );
+              expect(generatedAnswer).not.toBeNull();
+
+              expect(
+                generatedAnswer.classList.contains(
+                  'generated-answer__answer--collapsed'
+                )
+              ).toBe(true);
+              // expect(generatedAnswer.style.maxHeight).toEqual(
+              //   `${defaultAnswerHeight}px`
+              // );
+            });
+          });
+
+          describe('when the value is smaller than the valid range', () => {
+            it('should set the answer height with the fallback default value', async () => {
+              mockAnswerHeight = defaultAnswerHeight;
+              const element = createTestComponent({
+                ...defaultOptions,
+                collapsible: true,
+                maxCollapsedHeight: 100,
+              });
+              await flushPromises();
+
+              const generatedAnswer = element.shadowRoot.querySelector(
+                selectors.generatedAnswer
+              );
+              expect(generatedAnswer).not.toBeNull();
+
+              expect(
+                generatedAnswer.classList.contains(
+                  'generated-answer__answer--collapsed'
+                )
+              ).toBe(false);
+              // expect(generatedAnswer.style.maxHeight).toEqual(
```

Comment:
this is where I am trying to compare the height but its undefined. The element is defined though...

Reply:
BTW we never set the `maxHeight` property of the `style` of the element.

Reply:
We set the `--maxHeight` property of the style:

```
const styles = this.generatedAnswerElement?.style;
styles.setProperty('--maxHeight', `${this.maxCollapsedHeight}px`);
```

Meaning basically what gets rendered is this I think:

`<element class="generated-answer__answer" style="--maxHeight:250px">`

Reply:
Maybe try to use the `getPropertyValue` of the element?

https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/getPropertyValue

And make sure you call the `getPropertyValue` on the right element.

`this.template.querySelector('.generated-answer__answer');`

This is the element that gets set the `--maxHeight` property.